libs/math/src/math/linalg/blas/gemm_tn_novector.cpp

```diff
@@ -38,21 +38,21 @@ void Blas<S, Signature(_C <= _alpha, _A, _B, _beta, _C),
   std::size_t i;
   std::size_t j;
   if ((c.height() == 0) ||
-      ((c.width() == 0) || (((alpha == static_cast<Type>(0.0)) || (a.height() == 0)) &&
-                            (beta == static_cast<Type>(1.0)))))
+      ((c.width() == 0) || (((alpha == fetch::math::Type<Type>("0.0")) || (a.height() == 0)) &&
+                            (beta == fetch::math::Type<Type>("1.0")))))
   {
     return;
   }
 
-  if (alpha == static_cast<Type>(0.0))
+  if (alpha == fetch::math::Type<Type>("0.0"))
   {
-    if (beta == static_cast<Type>(0.0))
+    if (beta == fetch::math::Type<Type>("0.0"))
     {
       for (j = 0; j < c.width(); ++j)
       {
         for (i = 0; i < c.height(); ++i)
         {
-          c(i, j) = static_cast<Type>(0.0);
+          c(i, j) = fetch::math::Type<Type>("0.0");
```

Comment:
blas is a performance critical part, it should not have any string parser invoked anywhere